src/drivers/net/intel.c

```diff
@@ -674,6 +674,19 @@ static int intel_open ( struct net_device *netdev ) {
 		  INTEL_TCTL_COLD_DEFAULT );
 	writel ( tctl, intel->regs + INTEL_TCTL );
 
+	/* The Intel I210's packet buffer size registers only reset on power up.
+	 * If an operating system changes these but then the computer recieves a
+	 * reset signal without losing power, the registers will stay the same
+	 * (but be incompatible with other register defaults), thus making the
+	 * device unable to pass traffic.
+	 */
+	if ( intel->flags & INTEL_NO_PBSIZE_RST ) {
+		writel ( INTEL_I210_RXPBSIZE_DEFAULT, intel->regs + INTEL_E1000_RXPBS );
+		writel ( INTEL_I210_TXPBSIZE_DEFAULT, intel->regs + INTEL_E1000_TXPBS );
+		writel ( INTEL_I210_DTXMXPKTSZ,
+		         intel->regs + INTEL_I210_DTXMXPKTSZ_DEFAULT );
+	}
+
```

Comment:
The commit log message states that the `DTXMXPKTSZ` register **does** get reset as expected by a software reset.  Do we need to include code to set its value?

Reply:
No, not necessarily. I just included it since the incompatibility between those three registers were what was causing issues. In the latest revision I've removed the `DTXMXPKTSZ` defines